susemanager/bin/mgr-setup

```diff
@@ -199,75 +199,108 @@ cleanup_hostname() {
 
 exists_db() {
     PGNAME=$1
-    EXISTS=$(runuser - postgres -c 'psql -t -c "select datname from pg_database where datname='"'$PGNAME'"';"')
+    if [ $LOCAL_DB = 1 ]; then
+        EXISTS=$(runuser - postgres -c 'psql -t -c "select datname from pg_database where datname='"'$PGNAME'"';"')
+    else
+        EXISTS=$(echo "select datname from pg_database where datname='$PGNAME';" | (export PGPASSWORD=$EXTERNAL_DB_PASS; exec psql -t -U postgres -h $MANAGER_DB_HOST))
+    fi
     if [ "x$EXISTS" == "x $PGNAME" ] ; then
         return 0
     else
         return 1
     fi
 }
 
+exists_user() {
+    if [  $LOCAL_DB = 1 ] ; then
+        EXISTS=$($RUNUSER - postgres -c 'psql -t -c "select usename from pg_user where usename='"'$MANAGER_USER'"';"')
+    else
+        EXISTS=$(echo "select usename from pg_user where usename='$MANAGER_USER';" | (export PGPASSWORD=$EXTERNAL_DB_PASS; exec psql -t -U postgres -h $MANAGER_DB_HOST))
+    fi
+    if [ "x$EXISTS" == "x $MANAGER_USER" ] ; then
+        return 0
+    else
+        return 1
+    fi
+}
+
 setup_db_postgres() {
-    DATADIR="/var/lib/pgsql/data" 
-    systemctl --quiet enable postgresql 2>&1
-    if [[ "echo $(source /etc/os-release && echo ${ID_LIKE})" != *"suse"* ]]; then
-        # Create the PostgreSQL data folder, should it not exist.
-        if [ ! -f $DATADIR/PG_VERSION ]; then
-            rm -Rf ${DATADIR}
-            POSTGRES_LANG=en_US.UTF-8
-
-            # Define LC_CTYPE in postgres user profile. (Used during any DB creation)
-            PGHOME=$(getent passwd postgres | awk -F: '{print $6}')
-            . $PGHOME/.i18n 2>/dev/null
-            if [ -z $LC_CTYPE ]; then
-                grep "^LC_CTYPE" $PGHOME/.i18n > /dev/null 2>&1
+    if [ $LOCAL_DB = 1 ]; then
+        DATADIR="/var/lib/pgsql/data" 
```

Comment:
My feeling is, that we should re-think the DB setup implementation in `mgr-setup`.
We have now an external tool for the reportdb and also for the main db there exists an external tool (spacewalk-setup-postgresql) which we do not use.

Maybe we should combine this all in 1 setup tool.
This would make the code much cleaner.
CC @admd

Reply:
clarification question... are you suggesting this as a larger scope change for this PR or something that should be on the roadmap in the longer term?

Reply:
I think we need a refactoring, but not on this PR. Maybe we start this after this is in if time permits